src/ert/_c_wrappers/enkf/enkf_main.py

```diff
@@ -362,47 +473,74 @@ def sample_prior(
         # (this is a real code smell that we mute for now)
         if parameters is None:
             parameters = self._parameter_keys
-        state_map = storage.getStateMap()
 
         for parameter in parameters:
             config_node = self.ensembleConfig().getNode(parameter)
-            enkf_node = EnkfNode(config_node)
             if config_node.getUseForwardInit():
                 continue
-            impl_type = enkf_node.getImplType()
-            if impl_type == ErtImplType.GEN_KW:
-                gen_kw_node = enkf_node.asGenKw()
-                keys = [val[0] for val in gen_kw_node.items()]
+            impl_type = config_node.getImplementationType()
+            if impl_type == ErtImplType.FIELD:
+                for _, realization_nr in enumerate(active_realizations):
+                    init_file = config_node.get_init_file_fmt()
+                    if "%d" in init_file:
+                        init_file = init_file % realization_nr
+                    grid = xtgeo.grid_from_file(self.ensembleConfig().grid_file)
+                    props = xtgeo.gridproperty_from_file(
```

Comment:
This assumes that the files containing grid-properties have some extension like `GRDECL` and will fail if not.
`main` works even if the files have no such extension, for example, `PORO_0`, `PORO_1` etc. which makes this regression of sorts.
Thoughts @frode-aarstad  and @oyvindeide ?

Reply:
This is correct. We are adressing this in a new issue now that we store transformation, etc long with the field data